### PR TITLE
6 - Get comments for a particular article

### PR DESCRIPTION
**Pull request by HaylzRandom**
Created endpoint for getting comments for a particular article.

Unable to test for an article that does not exist.

Using promises to get the articleById and to get the comments and would assume that it would return message about article not existing but the comments promise resolves/rejects first 99.9% of the time so it states that no comments exist (which technically is true but not very user friendly)

Spoke to Poonam about it but not heard anything back and don't want to be stuck on a single test not passing due to race conditions

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -181,3 +181,60 @@ describe('/api/articles', () => {
...
+  */
+  test('GET:200 should return an array of comments for a given article', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body;
+
+        expect(comments).toHaveLength(11);
+        expect(comments).toBeSortedBy;
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: I'm not sure this works without being invoked with something, I know .toBeSorted by itself does but I'm not sure for .toBeSortedBy?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -181,3 +181,60 @@ describe('/api/articles', () => {
...
+        comments.forEach((comment) => {
+          expect(typeof comment.comment_id).toBe('number');
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: expect.objectContaining would be better and a bit clearer here than checking the type of all these properties

File: __tests__/app.test.js [hunk 3]
```diff
@@ -181,3 +181,60 @@ describe('/api/articles', () => {
...
+      });
+  });
+  test('GET:200 should return array of comments sorted by date in descending order', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body;
+
+        expect(comments).toBeSortedBy('created_at', {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:224]: perfect

File: __tests__/app.test.js [hunk 4]
```diff
@@ -181,3 +181,60 @@ describe('/api/articles', () => {
...
+          descending: true,
+        });
+      });
+  });
+  test('GET:404 should respond with appropriate message and status code when article exists but has no comments', () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: this should be a 200 not a 404 since it is a valid request that's completed without error

File: db/seeds/utils.js [hunk 5]
```diff
@@ -20,3 +23,13 @@ exports.formatComments = (comments, idLookup) => {
     };
   });
 };
+
+exports.checkExists = (table, column, value) => {
```
Comment (roisinoreilly) [inline db/seeds/utils.js:27]: really nicely done

File: models/comments.model.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+exports.getAllCommentsForArticle = (article_id) => {
+  const query = `
+    SELECT * FROM comments 
+    WHERE article_id = $1
+    ORDER BY created_at DESC;`;
+
+  return db.query(query, [article_id]).then(({ rows }) => {
+    if (rows.length === 0) {
+      return checkExists('articles', 'article_id', article_id).then(() => {
```
Comment (roisinoreilly) [inline models/comments.model.js:12]: perfectly done! (except it's a 404 not a 200 but otherwise really nice)

**Review by roisinoreilly — CHANGES_REQUESTED** [review]

**Review by Beth-McCormack — APPROVED** [review]

**Pull request merged**